tools/http3-test/src/lib.rs

```diff
@@ -0,0 +1,440 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+// IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+//! 🔧 HTTP/3 integration test utilities.
+//!
+//! This crate provides utilities to help integration tests against HTTP/3
+//! endpoints. Structures and methods can be combined with a [`quiche`]
+//! HTTP/3 client to run tests against a server. This client could be a
+//! binary or run as part of cargo test.
+//!
+//! ## Creating a test
+//!
+//! A test is an instance of [`Http3Test`], which consists of a set of
+//! [`Http3Req`] and a single [`Http3Assert`].
+//!
+//!
+//! Creating a single request:
+//!
+//! ```no_run
+//! let mut url = url::Url::parse("https://cloudflare-quic.com/b/get").unwrap();
+//! let mut reqs = Vec::new();
+//!
+//! reqs.push(http3-test::Http3Req::new("GET", &url, None, None));
+//! ```
+//!
+//! Assertions are used to check the received response headers and body
+//! against expectations. Each test has a [`Http3Assert`] which
+//! can access the received data. For example, to check the response
+//! status code is a 200 we could write the function:
+//!
+//! ```no_run
+//! fn assert_status(reqs: &[http3-test::Http3Req]) {
+//!     let status = reqs[0].resp_hdrs.iter().find(|&x| x.name() == ":status").unwrap();
+//!     assert_eq!(status.value(), "200");
+//! }
+//! ```
+//!
+//! However, because checking response headers is so common, for convenience
+//! the expected headers can be provided during [`Http3Assert`] construction:
+//!
+//! ```no_run
+//! let mut url = url::Url::parse("https://cloudflare-quic.com/b/get").unwrap();
+//! let mut reqs = Vec::new();
+//!
+//! let expect_hdrs = Some(vec![quiche::h3::Header::new(":status", "200")]);
+//! reqs.push(http3-test::Http3Req::new("GET", &url, None, expect_hdrs));
```

Comment:
errr, @ghedo so when I accepted your suggestion to rename the module from `http3test` to `http3-test` I had to update all the namespace usages, but now rustfmt wants to turn this line into 

```
reqs.push(http3 - test::Http3Req::new("GET", &url, None, expect_hdrs));
```

Reply:
Lol... I think yo need to replace `-` with `_` (to be consistent I guess we should always use `http3_test` instead of `http3-test` in code).